README_fr.md

```diff
@@ -0,0 +1,1641 @@
+
+### Je veux configurer une branche distante en tant qu'upstream pour une branche locale
+
+Vous pouvez configurer une branche distante en tant qu'upstream pour la branche locale actuelle en utilisant :
+
+```sh
+$ git branch --set-upstream-to [nomduremote]/[branche]
+# ou, en utilisant le raccourci :
+$ git branch -u [nomduremote]/[branche]
+```
+
+Pour configurer la branche distante en tant qu'upstream pour une autre branche locale :
+
+```sh
+$ git branch -u [nomduremote]/[branche] [branche-locale]
+```
+
+<a name="i-want-to-set-my-HEAD-to-track-the-default-remote-branch"></a>
+### Je veux configurer mon HEAD pour suivre la branche distante par défaut
+
+En vérifiant vos branches distantes, vous pouvez voir lesquelles d'entre-elles sont suivies par HEAD. Dans certains cas, ce n'est pas la branche désirée.
+
+```sh
+$ git branch -r
+  origin/HEAD -> origin/gh-pages
+  origin/master
+```
+
+Pour changer cela afin que `origin/HEAD` suive `origin/master`, vous pouvez lancer cette commande :
+
+```sh
+$ git remote set-head origin --auto
+origin/HEAD set to master
+```
+
+### J'ai fait des modifications sur la mauvaise branche
+
+Si vous avez fait des modifications non commitées et réalisez par la suite que vous êtes sur la mauvaise branche, remisez les modifications et appliquez-les sur la branche que vous voulez :
+
+```sh
+(mauvaise_branche)$ git stash
+(mauvaise_branche)$ git checkout &lt;bonne_branche&gt;
+(bonne_branche)$ git stash apply
+```
+
+## Rebaser et fusionner
+
+<a name="undo-rebase"></a>
+### Je veux annuler un rebase/merge
+
+Il se peut que vous avez fait une fusion ou un rebase sur votre branche actuelle avec une mauvaise branche, ou que vous vous êtes perdus ou n'arrivez pas à finir le processus de rebase ou de fusion. Git sauvegarde le pointeur de HEAD original dans une variable appelée ORIG_HEAD avant de faire des opérations dangereuses, afin qu'il soit plus facile de réinitialiser votre branche dans l'état avant le rebase ou la fusion :
+
+```sh
+(ma-branche)$ git reset --hard ORIG_HEAD
+```
+
+<a name="force-push-rebase"></a>
+### J'ai rebase, mais je ne veux pas pousser de force
+
+Malheureusement, vous devez pousser de force si vous souhaitez que ces modifications soient répercutées sur la branche distante. C'est parce que vous avez changé l'historique. La branche distante n'acceptera pas les modifications, sauf si vous poussez de force. C’est l’une des principales raisons pour lesquelles de nombreuses personnes utilisent un flux de travail à base de fusions plutôt qu'un flux de travail à base de rebase : les grandes équipes peuvent avoir des problèmes avec les développeurs qui poussent de force. Utilisez ceci avec prudence. Une façon plus sûre d’utiliser rebase est de ne pas du tout refléter vos modifications sur la branche distante, mais plutôt de procéder de la manière suivante :
```

Comment:
```suggestion
Malheureusement, vous devez pousser de force si vous souhaitez que ces modifications soient répercutées sur la branche distante. C'est parce que vous avez changé l'historique. La branche distante n'acceptera pas les modifications, sauf si vous poussez de force. C'est l'une des principales raisons pour lesquelles de nombreuses personnes utilisent un flux de travail à base de fusions plutôt qu'un flux de travail à base de rebase : les grandes équipes peuvent avoir des problèmes avec les développeurs qui poussent de force. Utilisez ceci avec prudence. Une façon plus sûre d'utiliser rebase est de ne pas du tout refléter vos modifications sur la branche distante, mais plutôt de procéder de la manière suivante :
```
French `’` there. It's no problem really, but it clashes with the rest of the document, which uses `'`.